.github/workflows/end_to_end_tests.yaml

```diff
@@ -0,0 +1,107 @@
+# (C) Copyright 2025, SECO Mind Srl
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: End to End tests
+
+on: [pull_request]
+env:
+  ASTARTE_API_HOSTNAME: api.autotest.astarte-platform.org
+  ASTARTE_BROKER_HOSTNAME: broker.autotest.astarte-platform.org
+  ASTARTE_ACTION_CERT: /usr/local/share/ca-certificates/astarte-autotest.crt
+  QEMU_GATEWAY_ADDR: 10.0.2.2
+
+jobs:
+  end-to-end-test:
+    runs-on: ubuntu-latest
+    concurrency:
+      group: end-to-end-test-${{ github.ref }}
+      cancel-in-progress: true
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        path: astarte-device-sdk-esp32
+    - name: Install ESP IDF
+      run: |
+        sudo apt-get update
+        sudo apt-get install git wget flex bison gperf python3 python3-pip python3-venv cmake ninja-build ccache libffi-dev libssl-dev dfu-util libusb-1.0-0
+        mkdir -p esp
+        cd esp
+        git clone -b release/v5.4 --recursive https://github.com/espressif/esp-idf.git
+        mkdir esp-tools
+        export IDF_TOOLS_PATH="$PWD/esp-tools"
+        cd esp-idf
+        ./install.sh all
+        echo "IDF_TOOLS_PATH=$IDF_TOOLS_PATH" >> "$GITHUB_ENV"
+        echo "IDF_PATH=$PWD" >> "$GITHUB_ENV"
```

Comment:
Why don't we use the idf container? For example `espressif/idf:release-v5.4`, so we can skip these installation step?

Reply:
Because the container is lacking some dependencies that are required by the end to end tests.